contracts/stake/src/contract.rs

```diff
@@ -283,6 +292,32 @@ impl StakingTrait for Staking {
         save_stakes(&env, &sender, &stakes);
     }
 
+    fn withdraw_rewards_chunks(env: Env, sender: Address, reward_token: Address, chunk_size: u32) {
+        env.storage()
+            .instance()
+            .extend_ttl(INSTANCE_LIFETIME_THRESHOLD, INSTANCE_BUMP_AMOUNT);
+
+        env.events().publish(("withdraw_rewards", "user"), &sender);
+
+        let mut stakes = get_stakes(&env, &sender);
+
+        // Calculate pending rewards in chunks
+        let (pending_rewards, last_reward_day) =
+            calculate_pending_rewards_chunked(&env, &reward_token, &stakes, chunk_size, None);
+
+        // Transfer pending rewards to the user
+        if pending_rewards > 0 {
+            token_contract::Client::new(&env, &reward_token).transfer(
+                &env.current_contract_address(),
+                &sender,
+                &pending_rewards,
```

Comment:
wouldn't it better if we use `env.contract_invoke` - should be better on the storage and (I'm not 100%) but it will avoid migration issues when we upgrade to next soroban-sdk